core/src/nucleus/validation/agent_entry.rs

```diff
@@ -0,0 +1,62 @@
+use crate::{
```

Comment:
This file is really the only new piece of core code. The rest is fixing tests and dealing with the macros